app/views/admin/users/index.html.haml

```diff
@@ -23,10 +23,11 @@
             %td
               = user.id
             %td
-              - if user.confirmed?
-                confirmed
-              - else
-                unconfirmed
+              - if can? :update, user
```

Comment:
When that is false, you are not showing anything, and it is impossible to know if the user is confirmed or not!
You can show the switch checkbox disabled. 
Also the action here is not update but `toggle_confirmation`, you should properly authorize. If you are not clear on how to do the authorization, find me on IRC to discuss this more.